### PR TITLE
Fix customer form validation from module

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | refer to https://github.com/PrestaShop/PrestaShop/issues/29234
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/29234
| Related PRs       | 
| How to test?      | refer to https://github.com/PrestaShop/PrestaShop/issues/29234
| Possible impacts? | 

Screenshot from 2022-08-23 13-40-17
Screenshot from 2022-08-23 13-40-11

### CODE REVIEW

**Review by kpodemski — APPROVED** [review]

**Review by MeKeyCool — APPROVED** [review]

File: classes/form/CustomerForm.php [hunk 1]
```diff
@@ -317,8 +317,14 @@ private function validateByModules()
...
                 $validatedCustomerFormFields = Hook::exec('validateCustomerFormFields', ['fields' => $formFields], $moduleId, true);
 
-                if (is_array($validatedCustomerFormFields)) {
-                    $this->formFields = array_merge($this->formFields, $validatedCustomerFormFields);
+                if (!is_array($validatedCustomerFormFields)) {
+                    continue;
+                }
+
+                foreach ($validatedCustomerFormFields as $name => $field) {
+                    if ($field instanceof FormFieldCore) {
```
Comment (rsoulard-prolaser) [inline classes/form/CustomerForm.php:325]: Replace FormFieldCore by FormField

**Review by rsoulard-prolaser — CHANGES_REQUESTED** [review]

File: classes/form/CustomerForm.php (hunk 1 above)
Comment (kpodemski) [reply classes/form/CustomerForm.php:325]: It should work for the `FormField` as well

Comment (jolelievre) [reply classes/form/CustomerForm.php:325]: `FormField` is supposed to extend `FormFieldCore` anyway so using the parent class for the check should also work, I don't think it's worth blocking the PR

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Review by florine2623 — APPROVED** [review]
Hello @zuk3975 ,

Thanks for the PR, tested on 8.0.x and launched automated tests.

Here's the result:

https://user-images.githubusercontent.com/16019289/187232436-3190db6f-025d-4a4c-b0c2-75d4e294b7d0.mov

It is looking good. No errors. Registration is OK. Checkout process works well after sign into customer account.

It is QA ✅

**Comment by atomiix** [comment]
Thanks @zuk3975 & @florine2623

**Pull request merged**